src/Samples/Sarif.WorkItems.Sample/ProgramStubs.cs

```diff
@@ -0,0 +1,233 @@
+            {
+                if (result.Guid == null)
+                {
+                    result.Guid = Guid.NewGuid().ToString();
+                }
+                guidToMatchedResultMap[result.Guid] = result;
+            }
+
+            var adoClient = new AzureDevOpsClient();
+            var filingTarget = new AzureDevOpsFilingTarget(adoClient);
+
+            var workItemFiler = new WorkItemFiler(
+                filingTarget,
+                filteringStrategy: new NewResultsFilteringStrategy(),
+                groupingStrategy: new OneResultPerWorkItemGroupingStrategy());
+
+            string logFilePath = Path.GetTempFileName();
+            File.WriteAllText(logFilePath, JsonConvert.SerializeObject(matchedLog));
+
+            // IMPORTANT! There's no facility defined yet for creating/configuring titles, area paths, etc.
+
+            try
+            {
+                IEnumerable<ResultGroup> resultGroups = workItemFiler.FileWorkItems(logFilePath).Result;
+                foreach (ResultGroup resultGroup in resultGroups)
+                {
+                    foreach (Result result in resultGroup.Results)
+                    {
+                        bool matched = guidToMatchedResultMap.TryGetValue(result.Guid, out Result originalResult);
+                        Debug.Assert(matched);
+                        originalResult.WorkItemUris = result.WorkItemUris;
+                    }
+                }
+            }
+            finally
+            {
+                if (File.Exists(logFilePath)) File.Delete(logFilePath);
+            }
+        }
+#endif
+
+        #region Stubs
+        private static SarifLog RetrieveCurrentSarif(string v)
+        {
+            string sarifText = GetResourceText("Sarif.Sdk.Sample.SampleTestFiles.Current.sarif");
+            return JsonConvert.DeserializeObject<SarifLog>(sarifText);
+        }
+
+        private static SarifLog RetrieveBaselineSarif(string v)
+        {
+            string[] resourceNames = typeof(Program).Assembly.GetManifestResourceNames();
+            Console.WriteLine(String.Join("\r\n", resourceNames));
+
+            string sarifText = GetResourceText("Sarif.Sdk.Sample.SampleTestFiles.Baseline.sarif");
+            return JsonConvert.DeserializeObject<SarifLog>(sarifText);
+        }
+
+        public static string GetResourceText(string resourceName)
+        {
+            string text = null;
+
+            using (Stream stream = typeof(Program).Assembly.GetManifestResourceStream(resourceName))
+            {
+                if (stream == null) { return null; }
+
+                using (StreamReader reader = new StreamReader(stream))
+                {
+                    text = reader.ReadToEnd();
+                }
+            }
+            return text;
+        }
+
+        private interface ISarifLogBaseliner
+        {
+            SarifLog CreateBaselinedSarifLog(SarifLog baselineSarif, SarifLog currentSarif, bool assignGuidsToNewResults);
```

Comment:
>assignGuidsToNewResults [](start = 97, length = 23)

Could use default argument (`assignGuidsToNewResults = false`) rather than overload -- I've never thought about which is preferable.

Reply:
Opinions vary but I'm persuaded by the argument that defaults get confusing when your API evolves to have many of them. The counterargument goes that with defaults, the caller is guaranteed that there is exactly one implementation and doesn't have to worry about bugs in the delegation from one overload to the next.

---
In reply to: [307081676](https://github.com/microsoft/sarif-sdk/pull/1589#discussion_r307081676) [](ancestors = 307081676)

Reply:
These API signatures are speculative, not crafted. I actually decided to remove this arg from the use case, which is what you should be looking at (i.e., the sample is the truth). Guids are something that s/be added or removed just as other data is handled.

---
In reply to: [307082989](https://github.com/microsoft/sarif-sdk/pull/1589#discussion_r307082989) [](ancestors = 307082989,307081676)